docker/documentation-site-builder/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM ruby:3.1-slim-bullseye as jekyll
+
+SHELL ["/bin/bash", "-eo", "pipefail", "-c"]
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    git \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /docs
+
+# First copy the Gemfile
+COPY docs/Gemfile /docs/
+COPY docs/Gemfile.lock /docs/
+
+RUN gem update --system && gem install jekyll && gem install just-the-docs && gem cleanup
```

Comment:
A few things here:

1- Don't update all gems because this makes the build not reproducible.
2- Install gems with `bundle install`, so you pick the contents from the `Gemfile`.

Reply:
Done